app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserMenuSignIn.kt

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import android.content.Context
+import android.view.View
+import android.widget.TextView
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat
+import mozilla.components.browser.menu.BrowserMenu
+import mozilla.components.browser.menu.item.BrowserMenuImageText
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+
+/**
+ * A menu item for displaying account information. The item calculates the label on every bind call,
+ * to provide each menu with the latest account manager state.
+ *
+ * @param label The function that provides the visible label of this menu item.
+ * @param imageResource ID of a drawable resource to be shown as icon.
+ * @param textColorResource Optional ID of color resource to tint the text.
+ * @param listener Callback to be invoked when this menu item is clicked.
+ */
+
+class BrowserMenuSignIn(
+    @ColorRes private val textColorResource: Int,
+    @DrawableRes imageResource: Int = R.drawable.ic_signed_out,
+    listener: () -> Unit = {},
+) : BrowserMenuImageText(
+    String(),
+    imageResource,
+    textColorResource = textColorResource,
+    listener = listener,
+) {
+    override fun bind(menu: BrowserMenu, view: View) {
+        super.bind(menu, view)
+        val textView = view.findViewById<TextView>(R.id.text)
+        textView.text = getLabel(textView.context)
+        textView.setTextColor(ContextCompat.getColor(textView.context, textColorResource))
+    }
+
+    /**
+     * Determines the proper label for the sign in button.  There are three states:
+     * user is signed-in, please sign-in, and we are not sure what's happening. The last state does
+     * occur when the menu is opened before the AccountManager has finished updating its state, or
+     * when there is no connection and AccountManager doesn't return us the profile because its cache
+     * has 2 minutes 'freshness':
+     * https://searchfox.org/mozilla-mobile/source/application-services/components/fxa-client/src/internal/profile.rs#41
```

Comment:
```suggestion
     * Returns the proper label for the sign in menu button.
```
I think the comments could be simplified here since the code are now just doing simple checks for whether or not the user is signed in or not. So, I could reasonably imagine that we don't have to care about the 3 states and especially this 'freshness' state where we are saying we don't know what's happening - it's just kinda concerning to say we aren't sure what's happening in the landed code.